src/core/utils/channel_manager.cpp

```diff
@@ -246,17 +351,26 @@ Error ChannelManager::RequestChannelSelect(bool aSkipQualityCheck)
 
     VerifyOrExit(!Get<Mle::Mle>().IsDisabled(), error = kErrorInvalidState);
 
-    VerifyOrExit(aSkipQualityCheck || ShouldAttemptChannelChange());
+    VerifyOrExit(aSkipQualityCheck || ShouldAttemptChannelChange(), error = kErrorAbort);
 
     SuccessOrExit(error = FindBetterChannel(newChannel, newOccupancy));
 
-    curChannel   = Get<Mac::Mac>().GetPanChannel();
+    curChannel = Get<Mac::Mac>().GetPanChannel();
+#if OPENTHREAD_CONFIG_CHANNEL_MANAGER_CSL_CHANNEL_SELECT_ENABLE
+    if (Get<Mac::Mac>().IsCslEnabled())
+    {
+        if (Get<Mac::Mac>().GetCslChannel() != 0)
+        {
+            curChannel = Get<Mac::Mac>().GetCslChannel();
+        }
+    }
+#endif
```

Comment:
```suggestion
#if OPENTHREAD_CONFIG_CHANNEL_MANAGER_CSL_CHANNEL_SELECT_ENABLE
    if (Get<Mac::Mac>().IsCslEnabled() && Get<Mac::Mac>().GetCslChannel() != 0)
    {
        curChannel = Get<Mac::Mac>().GetCslChannel();
    }
    else
#else
    {
        curChannel = Get<Mac::Mac>().GetPanChannel();
    }
#endif
```
We can avoid the `GetPanChannel()` call when CSL is enabled.